foo.sh

```
@@ -0,0 +1,657 @@
+    return 0
+  }
+  
+  # Install tools defined in .tool-versions
+  install_asdf_tools() {
+    if ! _exists asdf; then
+      error "ASDF not installed"
+      return 1
+    fi
+    
+    if [[ ! -f ~/.tool-versions ]]; then
+      error "~/.tool-versions not found"
```

Shellcheck (0.10.0)

[warning] Tilde does not expand in quotes. Use $HOME.

(SC2088)